web3-functions/pyth-oracle-w3f-priceIds/index.ts

```diff
@@ -0,0 +1,235 @@
+  let pythConfig: PythConfig | undefined;
+
+  let pythConfigStorage = JSON.parse(
+    (await storage.get("pythConfig")) ?? "{}"
+  ) as PythConfigStorage;
+
+  if (shouldFetchPythConfig(pythConfigStorage)) {
+    console.log("fetching pythConfig from gist");
+    try {
+      const gistDetails = await octokit.rest.gists.get({ gist_id: gistId });
+      const files = gistDetails.data.files;
+      if (!files) throw new Error(`No files in gist`);
+      for (const file of Object.values(files)) {
+        if (file?.filename === "config.yaml" && file.content) {
+          pythConfig = YAML.parse(file.content);
+          break;
+        }
+      }
+
+      if (!pythConfig) throw new Error(`No config.yaml loaded for PythConfig`);
+      pythConfigStorage = {
+        timestamp: Date.now() / 1000,
+        pythConfig: pythConfig,
+      };
+      const pythConfigStorageValue = JSON.stringify(pythConfigStorage);
+      console.log(`storing pythConfigStorageValue: ${pythConfigStorageValue}`);
+      await storage.set("pythConfig", pythConfigStorageValue);
+    } catch (err) {
+      const error = err as Error;
+      return {
+        canExec: false,
+        message: `Error fetching gist: ${error.message}`,
+      };
+    }
+  } else {
+    console.log("using pythConfig from storage");
+    pythConfig = pythConfigStorage.pythConfig;
+  }
+
+  console.log(`pythConfig: ${JSON.stringify(pythConfig, null, 2)}`);
+
+  const {
+    pythNetworkAddress,
+    priceServiceEndpoint,
+    validTimePeriodSeconds,
+    deviationThreshold,
+    priceIds,
+  } = pythConfig;
+
+  const pythnetwork = new Contract(
+    pythNetworkAddress,
+    pythAbi,
+    provider
+  ) as IPyth;
+
+  // Get Pyth price data
+  const connection = new EvmPriceServiceConnection(priceServiceEndpoint); // See Price Service endpoints section below for other endpoints
+
+  console.log(`checking priceIds: ${priceIds}`);
+  const check = (await connection.getLatestPriceFeeds(priceIds)) as any[];
+  console.log(`check: ${JSON.stringify(check, null, 2)}`);
+
+  const currentPrices: Map<string, IPRICE> = new Map();
+  check
+    .filter((c) => {
+      return c.price && c.price.price;
+    })
+    .forEach((c) => {
+      console.log(`c: ${JSON.stringify(c, null, 2)}`);
+      const price = {
+        price: +c.price.price,
+        timestamp: +c.price.publishTime,
+      };
+      currentPrices.set(addLeading0x(c.id), price);
+    });
+
+  console.log(
+    `currentPrices: ${JSON.stringify([...currentPrices.entries()], null, 2)}`
+  );
+
+  if (currentPrices.size != priceIds.length) {
+    console.log(
+      `currentPrices.priceIds: ${JSON.stringify(currentPrices.keys())}`
+    );
+    return { canExec: false, message: "Not all prices available" };
+  }
+
+  const lastPrices = (
+    await Promise.all(
+      [...currentPrices.keys()].map(async (priceId) => {
+        const price = JSON.parse(
+          (await storage.get(priceId)) ?? "{}"
+        ) as IPRICE;
```

Comment:
do you need IPRICE at all? seems like you could store whatever datatype the price service returns directly, and that would be simpler & more powerful.